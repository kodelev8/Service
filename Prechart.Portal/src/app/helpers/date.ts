import { Injectable } from "@angular/core";
import * as moment from "moment";

@Injectable({ providedIn: 'root' })
export class DateHelper {

  formatDate(date: any) {
    return moment(date).local().format('yyyy-MM-DD');
  }

}
