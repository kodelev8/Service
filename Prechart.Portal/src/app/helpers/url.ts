import { Injectable } from "@angular/core";
import { environment } from "src/environments/environment";

@Injectable({ providedIn: 'root' })
export class UrlHelper {
  format(subUrl: string, path: string) {
    var getUrl = this.getUrl(subUrl);

    if (environment.production) {
      var host = environment.apiUrl.replace('[[subdomain]]', getUrl);
      return `${host}${path}`;
    }
    else {
      var host = environment.apiUrl.replace('[[port]]', getUrl);
      return `${host}${path}`;
    }
  }

  getUrl(domain: string) {
    var value: string = '';
    switch (domain) {
      case 'document': value = environment.document;
        break;
      case 'belastingen': value = environment.belastingen;
        break;
      case 'user': value = environment.user;
        break;
      case 'loonheffings': value = environment.loonheffings;
        break;
      case 'person': value = environment.person;
        break;
      case 'werkgever': value = environment.werkgever;
        break;
      default: value = '0000';
        break;
    }
    return value;
  }
}
