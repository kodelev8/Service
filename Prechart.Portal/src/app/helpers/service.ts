import { HttpClient, HttpHeaders } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { AccountService } from "../shared/account/services/account.service";
import { UrlHelper } from "./url";
import {User} from "../shared/account/models/user";

@Injectable({ providedIn: 'root' })
export class ServiceHelper {
  headers: HttpHeaders;
  constructor(
    private http: HttpClient,
    private accountService: AccountService,
    private urlHelper: UrlHelper)
  {
    if (this.accountService.userValue) {
      var token = this.accountService.userValue;

      this.headers = new HttpHeaders({
        'Authorization': `Bearer ${token}`,
      });
    }
  }

  get(port: string, path: string) {
    var url = this.urlHelper.format(port, path);
    return this.http.get<any>(url, { headers: this.headers });
  }

  post(port: string, path: string, object: any) {
    var url = this.urlHelper.format(port, path);
    return this.http.post<any>(url, object, { headers: this.headers });
  }

  validateUser(port: string, path: string, object: any, options: any) {
    var url = this.urlHelper.format(port, path);
    return this.http.post<any>(url, object, options);
  }
}
