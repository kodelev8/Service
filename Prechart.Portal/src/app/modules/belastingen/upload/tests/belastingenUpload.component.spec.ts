import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { By } from '@angular/platform-browser';

import { MessageService } from 'primeng/api';
import { FileUploadModule } from 'primeng/fileupload';
import { CardModule } from 'primeng/card';
import { ToastModule } from 'primeng/toast';

import { BelastingenUploadComponent } from 'src/app/modules/belastingen/upload/belastingenUpload.component';
import { ServiceHelper } from '../../../../helpers/service';
import { AccountService } from '../../../../shared/account/services/account.service';
import { HttpClientModule } from '@angular/common/http';
import { RouterTestingModule } from '@angular/router/testing';

const uploadDocumentMock = jest.fn();

describe('BelastingenUploadComponent', () => {
  let fixture: ComponentFixture<BelastingenUploadComponent>;
  let component: BelastingenUploadComponent;
  let element: any;

  const getMockFn = jest.fn();
  let service: ServiceHelper;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        HttpClientTestingModule,
        FileUploadModule,
        CardModule,
        ToastModule,
        HttpClientModule,
        RouterTestingModule,
      ],
      declarations: [
        BelastingenUploadComponent,
      ],
      providers: [
        MessageService,
        AccountService
      ],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(BelastingenUploadComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
    fixture.detectChanges();
    service = TestBed.inject(ServiceHelper);

    service.get = getMockFn;
  });

  it('should create the Upload page', () => {  
    expect(component).toBeTruthy();
  });

  it('should render the Upload page correctly', () => {
    expect(element).toMatchSnapshot();
  });

  it('should have the Upload field', () => {
    const wrapper = fixture.debugElement.query(By.css("p-fileupload"));
    expect(wrapper).toBeTruthy();
    expect(element).toMatchSnapshot();
  });
});
