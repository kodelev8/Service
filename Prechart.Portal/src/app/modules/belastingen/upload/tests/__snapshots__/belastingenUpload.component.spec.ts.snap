// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BelastingenUploadComponent should have the Upload field 1`] = `
<div
  id="root2"
>
  <div
    class="div-center p-2"
  >
    <h3>
      Uploaden CSV
    </h3>
    <p-fileupload
      accept=".csv"
      class="p-element"
      name="fileUpload[]"
    >
      <div
        class="p-fileupload p-fileupload-advanced p-component"
      >
        <div
          class="p-fileupload-buttonbar"
        >
          <span
            class="p-ripple p-element p-button p-component p-fileupload-choose"
            pripple=""
            tabindex="0"
          >
            <input
              accept=".csv"
              multiple=""
              title=""
              type="file"
            />
            <span
              class="pi pi-plus p-button-icon p-button-icon-left"
            />
            <span
              class="p-button-label"
            >
              Choose
            </span>
          </span>
          <p-button
            class="p-element"
            type="button"
          >
            <button
              class="p-ripple p-element p-button p-component p-disabled"
              disabled=""
              pripple=""
              type="button"
            >
              
              <span
                aria-hidden="true"
                class="pi pi-upload p-button-icon p-button-icon-left"
              />
              
              <span
                aria-hidden="false"
                class="p-button-label"
              >
                Upload
              </span>
              
              
            </button>
          </p-button>
          
          <p-button
            class="p-element"
            type="button"
          >
            <button
              class="p-ripple p-element p-button p-component p-disabled"
              disabled=""
              pripple=""
              type="button"
            >
              
              <span
                aria-hidden="true"
                class="pi pi-times p-button-icon p-button-icon-left"
              />
              
              <span
                aria-hidden="false"
                class="p-button-label"
              >
                Cancel
              </span>
              
              
            </button>
          </p-button>
          
          
          
        </div>
        <div
          class="p-fileupload-content"
        >
          
          <p-messages
            class="p-element"
          >
            <div
              class="p-messages p-component"
              role="alert"
            >
              
              
              
              
            </div>
          </p-messages>
          
          
          
        </div>
      </div>
      
      
    </p-fileupload>
    <p-toast
      class="p-element"
      key="tr"
      position="top-right"
    >
      <div
        class="p-toast p-component p-toast-top-right"
      >
        
      </div>
    </p-toast>
  </div>
</div>
`;

exports[`BelastingenUploadComponent should render the Upload page correctly 1`] = `
<div
  id="root1"
>
  <div
    class="div-center p-2"
  >
    <h3>
      Uploaden CSV
    </h3>
    <p-fileupload
      accept=".csv"
      class="p-element"
      name="fileUpload[]"
    >
      <div
        class="p-fileupload p-fileupload-advanced p-component"
      >
        <div
          class="p-fileupload-buttonbar"
        >
          <span
            class="p-ripple p-element p-button p-component p-fileupload-choose"
            pripple=""
            tabindex="0"
          >
            <input
              accept=".csv"
              multiple=""
              title=""
              type="file"
            />
            <span
              class="pi pi-plus p-button-icon p-button-icon-left"
            />
            <span
              class="p-button-label"
            >
              Choose
            </span>
          </span>
          <p-button
            class="p-element"
            type="button"
          >
            <button
              class="p-ripple p-element p-button p-component p-disabled"
              disabled=""
              pripple=""
              type="button"
            >
              
              <span
                aria-hidden="true"
                class="pi pi-upload p-button-icon p-button-icon-left"
              />
              
              <span
                aria-hidden="false"
                class="p-button-label"
              >
                Upload
              </span>
              
              
            </button>
          </p-button>
          
          <p-button
            class="p-element"
            type="button"
          >
            <button
              class="p-ripple p-element p-button p-component p-disabled"
              disabled=""
              pripple=""
              type="button"
            >
              
              <span
                aria-hidden="true"
                class="pi pi-times p-button-icon p-button-icon-left"
              />
              
              <span
                aria-hidden="false"
                class="p-button-label"
              >
                Cancel
              </span>
              
              
            </button>
          </p-button>
          
          
          
        </div>
        <div
          class="p-fileupload-content"
        >
          
          <p-messages
            class="p-element"
          >
            <div
              class="p-messages p-component"
              role="alert"
            >
              
              
              
              
            </div>
          </p-messages>
          
          
          
        </div>
      </div>
      
      
    </p-fileupload>
    <p-toast
      class="p-element"
      key="tr"
      position="top-right"
    >
      <div
        class="p-toast p-component p-toast-top-right"
      >
        
      </div>
    </p-toast>
  </div>
</div>
`;
